pkg/asset/installconfig/platformcredscheck.go

```diff
@@ -0,0 +1,53 @@
+package installconfig
+
+import (
+	"fmt"
+
+	"github.com/gophercloud/utils/openstack/clientconfig"
+	"github.com/openshift/installer/pkg/asset"
+	awsconfig "github.com/openshift/installer/pkg/asset/installconfig/aws"
+	"github.com/openshift/installer/pkg/types/aws"
+	"github.com/openshift/installer/pkg/types/libvirt"
+	"github.com/openshift/installer/pkg/types/none"
+	"github.com/openshift/installer/pkg/types/openstack"
+)
+
+// PlatformCredsCheck is an asset that checks the platform credentials, asks for them or errors out if invalid
+// the cluster.
+type PlatformCredsCheck struct {
+}
+
+var _ asset.Asset = (*PlatformCredsCheck)(nil)
+
+// Dependencies returns the dependencies for PlatformCredsCheck
+func (a *PlatformCredsCheck) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&InstallConfig{},
+	}
+}
+
+// Generate queries for input from the user.
+func (a *PlatformCredsCheck) Generate(dependencies asset.Parents) error {
+	ic := &InstallConfig{}
+	dependencies.Get(ic)
+
+	var err error
+	platform := ic.Config.Platform.Name()
+	switch platform {
+	case aws.Name:
+		_, err = awsconfig.GetSession()
+	case libvirt.Name:
+	case none.Name:
+	case openstack.Name:
+		_, err = clientconfig.LoadCloudsYAML()
+	default:
+		err = fmt.Errorf("unknown platform type %q", platform)
```

Comment:
It's probably better if you do something like:

```go
		opts := new(clientconfig.ClientOpts)
		opts.Cloud = installConfig.Config.Platform.OpenStack.Cloud
		cloud, err := clientconfig.GetCloudFromYAML(opts)
		if err != nil {
			return err
		}
```

The above allows you to use the cloudname set in the installconfig, in case the environment variable is missing.

Reply:
thanks. Done.